modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -16,33 +16,41 @@ def create_a_record(fqdn, ip)
       if found = dns_find(fqdn)
         raise(Proxy::Dns::Collision, "#{fqdn} is already used by #{found}") if found != ip
       else
-        zone = match_zone(fqdn)
+        zone = match_zone(fqdn, enum_zones)
         msg = "Added DNS entry #{fqdn} => #{ip}"
         cmd = "/RecordAdd #{zone} #{fqdn}. A #{ip}"
         execute(cmd, msg)
       end
     end
 
-    # noop
-    def create_ptr_record(fqdn, ip)
-      found = dns_find(ip)
-      raise(Proxy::Dns::Collision, "#{ip} is already used by #{found}") if found && found != fqdn
-      true
+    def create_ptr_record(fqdn, ptr)
+      ip = ptr_to_ip ptr
+      if found = dns_find(ip)
+        raise(Proxy::Dns::Collision, "#{ptr} is already used by #{found}") unless fqdn == found
+      else
+        zone = match_zone(ptr, enum_zones)
+        msg = "Added PTR entry #{ptr} => #{fqdn}"
+        cmd = "/RecordAdd #{zone} #{ptr}. PTR #{fqdn}."
+        execute(cmd, msg)
+      end
     end
 
     def remove_a_record(fqdn)
       ip = dns_find(fqdn)
       raise Proxy::Dns::NotFound.new("Cannot find DNS entry for #{fqdn}") unless ip
```

Comment:
No point in doing this check: ms dns returns a success when deleting a non-existing record, foreman ignores 'not found' errors on delete. This only slowes down the response.